monkey/tests/unit_tests/infection_monkey/command_builders/test_windows_agent_command_builder.py

```diff
@@ -0,0 +1,199 @@
+    otp: str,
+    agent_otp_environment_variable: str,
+    agent_id: AgentID,
+):
+    windows_run_options = WindowsRunOptions(
+        agent_destination_path=AGENT_DESTINATION_PATH,
+        dropper_execution_mode=DropperExecutionMode.NONE,
+        shell=shell,
+    )
+
+    windows_agent_command_builder.build_run_command(windows_run_options)
+    actual_command = windows_agent_command_builder.get_command()
+
+    assert expected_otp_set in actual_command
+    assert agent_otp_environment_variable in actual_command
+    assert otp in actual_command
+    assert str(agent_id) in actual_command
+    assert MONKEY_ARG in actual_command
+    assert EXPECTED_AGENT_DESTINATION_PATH in actual_command
+
+
+@pytest.mark.parametrize(
+    "shell, expected_otp_set",
+    [
+        (WindowsShell.CMD, "set"),
+        (WindowsShell.POWERSHELL, "$env"),
+    ],
+)
+def test_build_run_command_dropper(
+    windows_agent_command_builder: IWindowsAgentCommandBuilder,
+    shell: WindowsShell,
+    expected_otp_set: str,
+    otp: str,
+    agent_otp_environment_variable: str,
+    agent_id: AgentID,
+):
+    windows_run_options = WindowsRunOptions(
+        agent_destination_path=AGENT_DESTINATION_PATH,
+        dropper_execution_mode=DropperExecutionMode.DROPPER,
+        shell=shell,
+        dropper_destination_path=DROPPER_DESTINATION_PATH,
+    )
+
+    windows_agent_command_builder.build_run_command(windows_run_options)
+    actual_command = windows_agent_command_builder.get_command()
+
+    assert expected_otp_set in actual_command
+    assert agent_otp_environment_variable in actual_command
+    assert otp in actual_command
+    assert str(agent_id) in actual_command
+    assert DROPPER_ARG in actual_command
+    assert EXPECTED_AGENT_DESTINATION_PATH in actual_command
+    assert EXPECTED_DROPPER_DESTINATION_PATH in actual_command
+
+
+@pytest.mark.parametrize(
+    "shell, expected_otp_set",
+    [
+        (WindowsShell.CMD, "set"),
+        (WindowsShell.POWERSHELL, "$env"),
+    ],
+)
+def test_build_run_command_script(
+    windows_agent_command_builder: IWindowsAgentCommandBuilder,
+    shell: WindowsShell,
+    expected_otp_set: str,
+    otp: str,
+    agent_otp_environment_variable: str,
+    agent_id: AgentID,
+):
+    windows_run_options = WindowsRunOptions(
+        agent_destination_path=AGENT_DESTINATION_PATH,
+        dropper_execution_mode=DropperExecutionMode.SCRIPT,
+        shell=shell,
+    )
+
+    windows_agent_command_builder.build_run_command(windows_run_options)
+    actual_command = windows_agent_command_builder.get_command()
+
+    assert expected_otp_set in actual_command
+    assert agent_otp_environment_variable in actual_command
+    assert otp in actual_command
+    assert EXPECTED_AGENT_DESTINATION_PATH in actual_command
+    assert str(agent_id) not in actual_command
+    assert DROPPER_ARG not in actual_command
+    assert MONKEY_ARG not in actual_command
+
+
+def test_build_run_command_unempty_download(
```

Comment:
Do we have a similar use case for linux? If so, I don't see a test for it

Reply:
We don't have similar use case because the code is different. Check the code of the concrete builder. Download is always powershell.